doc/lsst.ap.pipe/index.rst

```diff
@@ -19,10 +19,24 @@ Overview
 
    pipeline-overview
 
-.. .. _lsst.ap.pipe-using:
+.. _lsst.ap.pipe-using-gen2:
 
-Using lsst.ap.pipe
-==================
+Using lsst.ap.pipe in Gen 2
```

Comment:
I agree it is good to delineate Gen 2 and Gen 3 here, but please at least briefly explain (or link to) what that means, and ideally say something about how a user might choose which one to use.

Reply:
I also noticed the description at the top of this document describes ProcessCcdTask, ImageDifferenceTask, and AssociationTask as the three main things that AP Pipe does. Should ProcessCcdTask instead be broken down into its constituents since my understanding is it won't exist in Gen 3?

Reply:
> ideally say something about how a user might choose which one to use.

Except that the circumstances are kind of subject to change. Here's what I have so far, and I think it might be too inflexible:
> At present, the alert production pipeline is implemented using two separate frameworks that use incompatible data formats:
> 
> - `ApPipeTask` is an `lsst.pipe.base.CmdLineTask` that reads and writes data using the `lsst.daf.persistence` package.
>   This is the established "Gen 2" framework.
> - ``ApPipe`` is an `lsst.pipe.base.Pipeline` that reads and writes data using the `lsst.daf.butler` package.
>   This "Gen 3" framework is expected to be the only implementation in the future, but has limited support at the time of writing.

Any suggestions for how to make it less vulnerable to going out of date, but still specific enough for the reader to make an informed decision?

Reply:
These are exactly the kind of statements I was hoping for! You could omit "but has limited support at the time of writing." Instead of "that use incompatible data formats" you might say "that store and retrieve data from Butler repositories in different and incompatible ways"? But I think you absolutely captured the gist 👍